File: backend/src/main/java/io/github/daegwonkim/backend/entity/RentalItemImage.java
package io.github.daegwonkim.backend.entity;

import jakarta.persistence.*;
import lombok.Getter;

import java.util.UUID;

@Getter
@Entity
@Table(name = "rental_item_images")
public class RentalItemImage {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "UUID")
    private UUID id;

    @Column(name = "rental_item_id", columnDefinition = "UUID", nullable = false)
    private UUID rentalItemId;

    @Column(name = "image_url", nullable = false)
    private String imageUrl;
}

File: backend/domain/src/main/kotlin/io/github/daegwonkim/backend/entity/RentalItemImage.kt
package io.github.daegwonkim.backend.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "rental_item_images")
class RentalItemImage(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L,

    @Column(name = "rental_item_id", nullable = false)
    var rentalItemId: Long,

    @Column(nullable = false)
    var key: String,

    @Column(nullable = false)
    var sequence: Int
)